lib/experimental/assets/tinymce-proxy.js

```diff
@@ -0,0 +1,14 @@
+async function markAsRequiringTinymce() {
+	const expiration = new Date( 'Dec 31, 2099 23:59:59' );
+	document.cookie = `requiresTinymce=1; expires=${ expiration }; path=/`;
+	window.location.reload();
```

Comment:
We should be able to detect usage server side on the parse_blocks loop before hitting the client. Isn't that more straightforward?

Reply:
Well, I'm not sure about that - last time I checked, `parse_blocks()` would return an empty `blockName` for a classic block instance, and it returns the same if we have an empty space between blocks, too. Also, even if we were able to detect usage of a Classic block that way, we wouldn't be able to detect plugin usage that relies on the `wp.tinymce` global. Thoughts?

Reply:
Makes sense to me. I'm happy to try it out for detecting a classic block usage. Stay tuned.

I still think we're still going to need the cookie approach on the client side for instances of plugins that rely on the tinymce globals - there's no way to catch those from the server.

Reply:
what about setting a getter on `window.tinymce` that reloads the page with extra args to request it? no need to opt-in, and for those situations where we guessed wrong while being aggressive, we can correct that with a page refresh

```js
if (!('tinymce' in window)) {
	Object.defineProperty(window, 'tinymce', {
		get() {
			reloadPageWithMCE()
		}
	});
}
```

Reply:
(Ignore my accidental resolution of this thread.)

Reply:
> what about setting a getter on `window.tinymce` that reloads the page with extra args to request it? no need to opt-in, and for those situations where we guessed wrong while being aggressive, we can correct that with a page refresh

I'd like to see this tested too

Reply:
Sounds like a plan 👍 I wasn't very fond of the cookie solution either, so while I wanted to try with the block manager, this won't do (see [here](https://github.com/WordPress/gutenberg/pull/50387#issuecomment-1589081558) for more details), so I'll go with a query argument.

The only downside of that is that we might end up needing some extra effort to persist that query argument, but then again, we won't have to worry about cleaning it up either, so it might end up being quite a solid approach.

Will work on it and update y'all.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/50387/commits/896f8cf5a8b993dd409469af249a70fdc97c0467 to use a query argument instead of a cookie. Other than not persisting the value, this approach appears to be working fine - let me know what you think @dmsnell @mcsf